### ISSUE TITLE
Option to fail validation if the sequence of fields do not match

### ISSUE THREAD

**Issue by AbhijeetPande** [pydantic/pydantic#4587]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this feature is missing

### Description

I propose adding an option to fail validation if the sequence of fields do not match. Please view example below:

```
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


class User(BaseModel):
    id: int
    name = 'John Doe'
    signup_ts: Optional[datetime] = None
    friends: List[int] = []


external_data_1 = {'id': '123', 'signup_ts': '2017-06-01 12:22', 'friends': [1, '2', b'3']}
external_data_2 = {'signup_ts': '2017-06-01 12:22', 'id': '123', 'friends': [1, '2', b'3']}

# notice how external_data_2 has an order that does not follow the sequence of fields as declared in the pydantic class

user_1 = User(**external_data_1)
user_2 = User(**external_data_2)
print(user_1)
# > User id=123 name='John Doe' signup_ts=datetime.datetime(2017, 6, 1, 12, 22) friends=[1, 2, 3]
print(user_2)
# > User id=123 name='John Doe' signup_ts=datetime.datetime(2017, 6, 1, 12, 22) friends=[1, 2, 3]
```
User_2 is treated as if it is the exact same as User_1
but the two dicts are not the same
```
if (list(external_data_1) == list(external_data_2)):
    print("The two dicts are the same")
else:
    print("The two dicts are not the same")
# > The two dicts are not the same
```
is it possible to fail validation for dictionaries that do not follow the correct order of fields? 

Something like this: 
```
class User_strict_order(BaseModel, enforce_order=True):
id: int
name = 'John Doe'
signup_ts: Optional[datetime] = None
friends: List[int] = []
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [X] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment]
I don't think we want to add this as a built in feature, but you can accumplish it with a custom [root validator](https://pydantic-docs.helpmanual.io/usage/validators/#root-validators)
